packages/deployments/contracts/contracts/core/connext/facets/SwapAdminFacet.sol

```diff
@@ -176,6 +185,35 @@ contract SwapAdminFacet is BaseConnextFacet {
     emit SwapInitialized(_key, entry, msg.sender);
   }
 
+  /**
+   * @notice remove Swap Struct for key
+   *
+   * @param _key the hash of the canonical id and domain for token
+   */
+  function removeSwap(bytes32 _key) external onlyOwnerOrAdmin {
+    uint256 numPooledTokens = s.swapStorages[_key].pooledTokens.length;
+
+    if (numPooledTokens == 0) revert SwapAdminFacet__removeSwap_notInitialized();
+
+    for (uint256 i; i < numPooledTokens; ) {
+      if (s.swapStorages[_key].balances[i] == 0) {
```

Comment:
Shouldn't it be the other way around?
e.g. only allow removal of swap when all the balances are 0?
If the balances are >0 then the tokens would be lost.

Reply:
Yes, correct!

Reply:
But there is another problem here. Once swap pool is initialized and liquidity added, it is very hard to make all balances zero. It means that in this case, swap pool can't be removed!
Do you have any idea for this?

Reply:
Remove all total supply, can make balances zero. anyway fixed, please check again

Reply:
Yes, correct, but `addSwapLiquidity` is not owner function, so anyone can add liquidity to swap pool. In this case, someone doesn't remove their liquidity from swap pool, owner can't remove that swap. I think that would be problem and we need a `forceRemoveSwap` function.

Reply:
we could take a similar approach as 

in that case:
- `forceRemoveSwap` (rename to `disableSwap`) would set a flag that indicates whether it is usable
- `initializeSwap` could still be gated to admins, no need to leave that open

then you would be able to safely remove liquidity when the pool is disabled. you wouldn't be able to clean up the storage quite as well, though maybe that could be a separate function (and is called with some enforced delay from `disableSwap`). i think that may be overkill.

Reply:
That seems like a solid approach.
When the swap is disabled it can't be used of course, i presume the system should use an external amm then.

Reply:
we can add a functionality so that Connext Diamond can still let liquidity providers to remove their balance from a disabled AMM.

Reply:
what @gpersoon mentioned can make it work. Always use an external AMM. The liquidity providers deposit tokens directly to that AMM. Now Connext Diamond can just connect and disconnect from it without worrying about the token balance in AMM.